src/pages/ViewProgramme.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from "react";
+import { client } from "~/api";
+
+interface Programme {
+    id: string;
+    created_at: string;
+    updated_at: string;
+    name: string;
+    degree_type: string;
+    credits: string;
+    courses: string[];
+}
+
+interface ProgrammeResponse {
+    count: number;
+    next: string | null;
+    previous: string | null;
+    results: Programme[];
+}
+
+interface Course {
```

Comment:
I talked with backend they will add the year information to the course because we need for the filtering/highlight information. If you want to add it, it could be useful. Otherwise, we can add it on the next edit